aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/internal/ServiceUtils.java

```diff
@@ -346,7 +346,7 @@ public static boolean createParentDirectoryIfNecessary(final File file) {
         if (parentDirectory == null || parentDirectory.exists()) {
             return false;
         }
-        if (parentDirectory.mkdirs()) {
+        if (parentDirectory.mkdirs() || parentDirectory.isDirectory()) {
             return true;
```

Comment:
The return-type here is never actually used and doesn't really add any value; I propose we change the signature of this method to return void instead: 

```java
public static void createParentDirectoryIfNecessary(final File file) {
    final File parentDirectory = file.getParentFile();
    if (parentDirectory == null || parentDirectory.exists() || parentDirectory.mkdirs()) {
        return;
    }

    if (!parentDirectory.exists()) {
        throw new AmazonClientException("Unable to create directory in the path: " + parentDirectory.getAbsolutePath());
    }
}
```